landscaper/pkg/controlplane/apis/imports/types_apiserver.go

```diff
@@ -0,0 +1,438 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imports
+
+import (
+	hvpav1alpha1 "github.com/gardener/hvpa-controller/api/v1alpha1"
+	landscaperv1alpha1 "github.com/gardener/landscaper/apis/core/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	apiserverv1 "k8s.io/apiserver/pkg/apis/apiserver/v1"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	apiserverconfigv1 "k8s.io/apiserver/pkg/apis/config/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// GardenerAPIServer contains the configuration of the Gardener Aggregated API Server
+type GardenerAPIServer struct {
+	// DeploymentConfiguration contains optional configurations for
+	// the deployment of the Gardener API server
+	DeploymentConfiguration *APIServerDeploymentConfiguration
+	// ComponentConfiguration contains optional configurations for
+	// the Gardener Extension API server
+	ComponentConfiguration APIServerComponentConfiguration
+}
+
+// APIServerDeploymentConfiguration contains certain configurations for the deployment
+// of the Gardener Extension API server
+type APIServerDeploymentConfiguration struct {
+	// CommonDeploymentConfiguration contains common deployment configurations
+	// Defaults:
+	//  Resources: Requests (CPU: 100m, memory 100Mi), Limits (CPU: 300m, memory: 256Mi)
+	CommonDeploymentConfiguration
+	// LivenessProbe allows to overwrite the default liveness probe
+	// Defaults:
```

Comment:
I do not see any defaulting here. Is the defaulting added in the external type or do you plan to add the defaulting in the helm chart?

Reply:
I will add the defaulting in the external type. 
However, I am planning not to duplicate defaulting, which is already done in the helm chart.
At least that is currently the plan, but not 100% sure if this is how it should be done.

WDYT also @rfranzke

Reply:
> However, I am planning not to duplicate defaulting
Yes that was the background of my question. I think defaulting in different places will result in confusion and could be hard to debug.

Reply:
Good point, how was it done for the gardenlet landscaper component?

Reply:
defaulting was not duplicated.

In general, I wanted to make the deployment as easy as possible.
If there is something reasonable to be defaulted / generated which is currently not (e.g the TLS certificates and CA Bundles) then I'll add that.

Also for hvpa, if the user does not explicitly configure it,  I want to only enable it if the hvpa CRD exists. Otherwise hvpa will not be enabled per default.